services/transactionpool/add_new_transaction.go

```diff
@@ -40,20 +40,38 @@ func (s *service) AddNewTransaction(input *services.AddNewTransactionInput) (*se
 		return s.anEmptyReceipt(), err
 	}
 
-	s.gossip.BroadcastForwardedTransactions(&gossiptopics.ForwardedTransactionsInput{
-		Message: &gossipmessages.ForwardedTransactionsMessage{
-
-			SignedTransactions: []*protocol.SignedTransaction{input.SignedTransaction},
-		},
-	})
-
 	s.reporting.Info("adding new transaction to the pool", instrumentation.Stringable("transaction", input.SignedTransaction))
+	if _, err := s.pendingPool.add(input.SignedTransaction, s.config.NodePublicKey()); err != nil {
+		return nil, err
+
+	}
 	s.pendingTransactions <- input.SignedTransaction //TODO remove this
-	s.pendingPool.add(input.SignedTransaction)
+
+	//TODO batch
+	s.forwardTransaction(input.SignedTransaction)
 
 	return &services.AddNewTransactionOutput{}, nil
 }
 
+func (s *service) forwardTransaction(tx *protocol.SignedTransaction) error {
+	// TODO sign
+	//sig, err := signature.SignEd25519(s.config.NodePrivateKey(), signedData)
+	//if err != nil {
+	//	return nil, err
+	//}
+
+	_, err := s.gossip.BroadcastForwardedTransactions(&gossiptopics.ForwardedTransactionsInput{
+		Message: &gossipmessages.ForwardedTransactionsMessage{
+			SignedTransactions: []*protocol.SignedTransaction{tx},
```

Comment:
should it be an array here? (this is the copy i talked about, its not a copy, missed that it is an array)